share/functions/__fish_git_prompt.fish

```diff
@@ -471,14 +471,20 @@ end
 set -g ___fish_git_prompt_status_order stagedstate invalidstate dirtystate untrackedfiles
 
 function __fish_git_prompt_informative_status
-
     set -l changedFiles (command git diff --name-status | cut -c 1-2)
     set -l stagedFiles (command git diff --staged --name-status | cut -c 1-2)
+    set -l untrackedfiles 0
 
     set -l dirtystate (math (count $changedFiles) - (count (echo $changedFiles | grep "U")))
     set -l invalidstate (count (echo $stagedFiles | grep "U"))
     set -l stagedstate (math (count $stagedFiles) - $invalidstate)
-    set -l untrackedfiles (count (command git ls-files --others --exclude-standard))
+
+    # Untracked files may cause slow prompt generation.
+    # For this reason, it should be possible to avoid counting the files
+    # by setting __fish_git_prompt_showuntrackedfiles explicitly to "false".
+    if test "$__fish_git_prompt_showuntrackedfiles" != false
```

Comment:
Yes, this should also use the git config option ("bash.showUntrackedFiles").

However, I think for the "informative" prompt it should default to showing it, i.e. if neither our variable nor the git option are explicitly set to false it should count untracked files.

As a sidenote, I'm not quite a fan of the zoo of configuration variables we're using here, but so far I've not made any progress in finding a better design.